# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/StoreBuffer.h
# Commit: 982a80b202c3
# Full Hash: 982a80b202c370876dafc72ac31be7a12ebeae58
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2021-01-08 03:45:03
# Regressor Bug: 1681533
# File Overlap Count: 2
# Description:
#   Bug 1681533 - Only collect the nursery when necessary in major GC r=sfink
#   
#   This rearranges a few things so that we can run a minor GC inside a major GC
#   slice, so we don't have to be conservative about collecting the nursery at the
#   start of a slice.
# ==============================================================================

diff -r 83f3726c41f9 -r 982a80b202c3 js/src/gc/StoreBuffer.h
--- a/js/src/gc/StoreBuffer.h	Thu Jan 07 15:57:41 2021 +0000
+++ b/js/src/gc/StoreBuffer.h	Mon Jan 04 17:01:50 2021 +0000
@@ -480,6 +480,7 @@
   void disable();
   bool isEnabled() const { return enabled_; }
 
+  bool isEmpty() const;
   void clear();
 
   const Nursery& nursery() const { return nursery_; }